frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/TopicResource.java

```diff
@@ -0,0 +1,169 @@
+        } catch (Exception ex) {
+            log.error("Failed to create topic: " + name + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/unavailable_partitions")
+    public Response unavailablePartitions() {
+        try {
+            JSONObject obj = cmdExecutor.unavailablePartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/under_replicated_partitions")
+    public Response underReplicatedPartitions() {
+        try {
+            JSONObject obj = cmdExecutor.underReplicatedPartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @PUT
+    @Path("/operations/{type}")
+    public Response operationOnTopic(
+            @QueryParam("type") String type,
+            @PathParam("name") String topicName,
+            @QueryParam("key") String key,
+            @QueryParam("value") String value,
+            @QueryParam("partitions") String partitions,
+            @QueryParam("messages") String messages) {
+        try {
+            JSONObject result = null;
+            List<String> cmds = null;
+            if (type == null) {
+                result = new JSONObject();
+                result.put("Error", "Must designate an 'operation'.  " +
+                        "Possibles operations are [producer-test, delete, partitions, config, deleteConfig].");
+            } else {
+                switch (type) {
+                    case "producer-test":
+                        int messageCount = Integer.parseInt(messages);
+                        result = cmdExecutor.producerTest(topicName, messageCount);
+                        break;
+                    case "partitions":
+                        cmds = Arrays.asList("--partitions", partitions);
+                        result = cmdExecutor.alterTopic(topicName, cmds);
+                        break;
+                    default:
+                        result = new JSONObject();
+                        result.put("Error", "Unrecognized operation: " + type);
+                        break;
+                }
+            }
+            return Response.ok(result.toString(), MediaType.APPLICATION_JSON).build();
+
+        } catch (Exception ex) {
+            log.error("Failed to perform operation: " + type + " on Topic: " + topicName + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @DELETE
+    @Path("/delete/{name}")
```

Comment:
Since the `DELETE` method makes this unambiguous, the path should be `/{topic}` for RESTfulness.